src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/EditProjectFileCommand.cs

```diff
@@ -115,22 +147,26 @@ private void TextDocument_FileActionOccurred(object sender, TextDocumentFileActi
             }));
         }
 
-        private async Task<string> GetFileAsync(string projectFileName)
+        private async Task SetupFileAsync()
+        {
+            _tempProjectPath = GetTempFileName(Path.GetFileName(_unconfiguredProject.FullPath));
```

Comment:
Everything still works. However, the name on the buffer does not change, it's still the old file name. To really fix that, we probably want to refactor out the state machine, as it'll start getting quite complex if we do it in here.